backend/src/main/java/org/eclipse/tractusx/semantics/registry/utils/ShellSpecification.java

```diff
@@ -34,22 +41,49 @@ public class ShellSpecification<T> implements Specification<T> {
 
    private final String sortFieldName;
    private final ShellCursor shellCursor;
+   private final String tenantId;
+   private final String owningTenantId;
+   private final String publicWildcardPrefix;
+   private final List<String> publicWildcardAllowedTypes;
 
    @Override
    public Predicate toPredicate( Root<T> root, CriteriaQuery<?> cq, CriteriaBuilder criteriaBuilder ) {
-      return applyFilter( root,cq, criteriaBuilder );
+      return applyFilter( root, cq, criteriaBuilder );
    }
 
-   private Predicate applyFilter( Root<T> root,CriteriaQuery<?> cq, CriteriaBuilder criteriaBuilder ) {
-      if(root.toString().contains( "Shell" )){
-         var searchValue = shellCursor.getShellSearchCursor();
+   private Predicate applyFilter( Root<T> root, CriteriaQuery<?> cq, CriteriaBuilder criteriaBuilder ) {
+      if ( root.toString().contains( "Shell" ) ) {
+         Instant searchValue = shellCursor.getShellSearchCursor();
          cq.orderBy( criteriaBuilder.asc( criteriaBuilder.coalesce( root.get( sortFieldName ), Instant.now() ) ) );
-         return criteriaBuilder.greaterThan( root.get( sortFieldName ), searchValue );
-      }else{
-         var searchValue = shellCursor.getSubmodelSearchCursor();
+
+         if ( owningTenantId.equals( tenantId ) ) {
+            return criteriaBuilder.greaterThan( root.get( sortFieldName ), searchValue );
+         }
+
+         return getAllShellsPredicate( root, cq, criteriaBuilder, searchValue );
+      } else {
+         UUID searchValue = shellCursor.getSubmodelSearchCursor();
          cq.orderBy( criteriaBuilder.asc( criteriaBuilder.coalesce( root.get( sortFieldName ),
-               UUID.fromString( "00000000-0000-0000-0000-000000000000" )) ) );
+               UUID.fromString( "00000000-0000-0000-0000-000000000000" ) ) ) );
          return criteriaBuilder.greaterThan( root.get( sortFieldName ), searchValue );
       }
    }
+
+   private Predicate getAllShellsPredicate( Root<T> root, CriteriaQuery<?> cq, CriteriaBuilder criteriaBuilder, Instant searchValue ) {
```

Comment:
May be here instead of giving hard coded field names, you can use something like https://stackoverflow.com/questions/13192734/getting-a-property-field-name-using-getter-method-of-a-pojo-java-bean
Using @FieldNameConstants from lombok will provide you field names as string.